src/module.c

```diff
@@ -375,6 +376,19 @@ JL_DLLEXPORT jl_value_t *jl_binding_owner(jl_module_t *m, jl_sym_t *var)
     return (jl_value_t*)b->owner;
 }
 
+// get type of binding m.var, without resolving the binding
+JL_DLLEXPORT jl_value_t *jl_binding_type(jl_module_t *m, jl_sym_t *var)
+{
+    JL_LOCK(&m->lock);
+    jl_binding_t *b = (jl_binding_t*)ptrhash_get(&m->bindings, var);
+    if (b == HT_NOTFOUND || b->owner == NULL)
+        b = using_resolve_binding(m, var, NULL, 0);
```

Comment:
how does this work for imported bindings, once we know the owner? I don't think I see any tests for nested modules and/or using/export/imports

Reply:
Yes, I should add some tests. I believe this should already read the type of the correct binding, but I'll double check. Of course, there's still the same purity issue discussed above.